emcc.py

```diff
@@ -853,6 +853,14 @@ def phase_setup(options, state, newargs):
   if settings.SHARED_MEMORY:
     settings.BULK_MEMORY = 1
 
+  if '-mbulk-memory' not in newargs and '-mno-bulk-memory' not in newargs:
+    print('newargs bulk')
+    if feature_matrix.caniuse(feature_matrix.Feature.BULK_MEMORY):
+      newargs += ['-mbulk-memory']
+      settings.BULK_MEMORY = 1
+    else:
+      newargs += ['-mno-bulk-memory']
```

Comment:
The code in link.py should probably be moved here:
```
if not settings.BULK_MEMORY:                                                                       
  settings.BULK_MEMORY = feature_matrix.caniuse(feature_matrix.Feature.BULK_MEMORY)  
```

Then once the setting is determines we should set the command line flag based on that.

Reply:
Does the code here in emcc.py run at link time too? In that case we could maybe just delete the code in link.py entirely?
Or would we still need it for the case where emcc.py isn't used at all?

Reply:
Yes, emcc.py runs at both compile and link time.   If code needs to run at both compile and line time it should live there.  If its only needs to run at link time it should live in link.py

Reply:
Can we somehow consolidate the different places where `BULK_MEMORY` is set and `-mno-bulk-memory` / `-mbulk-memory` are added.

At the very least I don't think we need the caniuse in link.py and in emcc.py both.

Perhaps this block along with the block above for handling shared memory should just set `BULK_MEMORY`?   Then we can use the `BULK_MEMORY` setting along to set the command line flag?